### ISSUE TITLE
Incomplete Dependency Parsing

### ISSUE THREAD

**Issue by lskillen** [xolox/python-deb-pkg-tools#12]
Hello!  First of all, thank you for providing deb_pkg_tools, it's a library that we find immensely useful. :-)

This is just a quick query with regards to support for the "Recommends" (and other) control field when parsing dependencies, as it appears that it currently isn't, based on the `DEPENDS_LIKE_FIELDS` list:

```
DEPENDS_LIKE_FIELDS = (
    'Breaks',
    'Build-Depends',
    'Conflicts',
    'Depends',
    'Pre-Depends',
    'Provides',
    'Replaces',
    'Suggests',
)
```

This caused a problem for us when expecting the *Recommends* field to be a parsed relationship set, and was unexpectedly a string instead (unparsed).  The [Debian policy documentation](https://www.debian.org/doc/debian-policy/ch-relationships.html) mentions control fields in various sections that probably should belong in this list, such as:

- [Section 7.1](https://www.debian.org/doc/debian-policy/ch-relationships.html):
  - Recommends
- [Section 7.2](https://www.debian.org/doc/debian-policy/ch-relationships.html):
  - Enhances
- [Section 7.7](https://www.debian.org/doc/debian-policy/ch-relationships.html):
  - Build-Depends
  - Build-Depends-Indep
  - Build-Depends-Arch
  - Build-Conflicts
  - Build-Conflicts-Indep
  - Build-Conflicts-Arch
- [Section 7.8](https://www.debian.org/doc/debian-policy/ch-relationships.html):
  - Built-Using

An example of a debian package / control file that has *Recommends* is the [virtualbox-5.1 deb file](http://download.virtualbox.org/virtualbox/5.1.14/virtualbox-5.1_5.1.14-112924~Ubuntu~trusty_i386.deb):

> Package: virtualbox-5.1
Version: 5.1.14-112924~Ubuntu~trusty
Architecture: i386
Maintainer: Oracle Corporation <info@virtualbox.org>
Installed-Size: 168128
Pre-Depends: debconf (>= 1.1) | debconf-2.0
Depends: libc6 (>= 2.15), libcurl3 (>= 7.16.2), libdevmapper1.02.1 (>= 2:1.02.20), libfontconfig1 (>= 2.9.0), libfreetype6 (>= 2.2.1), libgcc1 (>= 1:4.1.1), libgl1-mesa-glx | libgl1, libglib2.0-0 (>= 2.12.0), libice6 (>= 1:1.0.0), libpng12-0 (>= 1.2.13-4), libpython2.7 (>= 2.7), libsdl1.2debian (>= 1.2.11), libsm6, libssl1.0.0 (>= 1.0.0), libstdc++6 (>= 4.6), libvpx1 (>= 1.0.0), libx11-6, libx11-xcb1, libxcb1 (>= 1.8), libxcursor1 (>> 1.1.2), libxext6, libxinerama1, libxml2 (>= 2.7.4), libxmu6, libxrender1, libxt6, zlib1g (>= 1:1.1.4), psmisc, adduser
Recommends: libasound2, libpulse0, libsdl-ttf2.0-0, kmod | kldutils | module-init-tools, linux-headers-generic | linux-headers-generic-pae | linux-headers-686-pae | linux-headers-amd64 | linux-headers-2.6-686 | linux-headers-2.6-amd64 | linux-headers, linux-image, gcc, make | build-essential | dpkg-dev, binutils, pdf-viewer, libgl1
Conflicts: virtualbox, virtualbox-guest-additions-iso, virtualbox-ose
Replaces: virtualbox
Provides: virtualbox
Section: contrib/misc
Priority: optional
Description: Oracle VM VirtualBox
VirtualBox is a powerful PC virtualization solution allowing you to run a
wide range of PC operating systems on your Linux system. This includes
Windows, Linux, FreeBSD, DOS, OpenBSD and others. VirtualBox comes with a broad
feature set and excellent performance, making it the premier virtualization
software solution on the market.
Python-Version: 2.7

We'd be fairly happy to contribute a PR if you agree that these should be parsed.  Thanks!

**Comment by xolox** [comment]
Hi Lee and thanks for the feedback!

> First of all, thank you for providing deb_pkg_tools, it's a library that we find immensely useful. :-)

That is always good to hear, I'm glad that you find it useful.

> We'd be fairly happy to contribute a PR if you agree that these should be parsed. Thanks!

I never bothered adding `Recommends` and `Suggests` because I've never needed to use these fields, but I don't see any negative impact from adding these to `DEPENDS_LIKE_FIELDS`.

The `Build-*` fields were never added to `DEPENDS_LIKE_FIELDS` because my original use case(s) for deb-pkg-tools focused only on control fields commonly found in binary package archives, but again I don't see any problem with adding these.

Long story short: A pull request is definitely welcome, if not then I will likely pick this up myself soon (TM).

**Comment by xolox** [comment]
Shortly after posting my previous message here, I realized that I never implemented a proper cache invalidation technique for the new package cache that was released in version 3.0 (which was added to support multiple simultaneous cache readers and writers without blocking). I was going to suggest:

> Please make sure to wipe the cache if this has the potential to bite.

But on second thought, not implementing a cache invalidation technique is lame, so I added support for cache invalidation in release 4.2. This should ensure that cached package metadata is invalidated when it contains unparsed `Suggests` / `Recommends`/ etc. fields.
